test/tsk/hashdb/test_binsrch_index.cpp

```diff
@@ -0,0 +1,94 @@
+#include "tsk/base/tsk_os.h"
+#include "tsk/hashdb/tsk_hashdb_i.h"
+#include "catch.hpp"
+
+void test_hdb_binsrch_idx_init_hash_type_info(
+    const TSK_TCHAR *db_name,
+    TSK_HDB_HTYPE_ENUM htype,
+    int expected_return,
+    int expected_hash_len,
+    const TSK_TCHAR *expected_idx_fname,
+    const TSK_TCHAR *expected_idx_idx_fname)
+{
+    TSK_HDB_BINSRCH_INFO hdb_binsrch_info;
+
+    /**
+     * setup values:
+     * normally, this would be done as a part of opening the hashdb, but not 
+     * for this unit test
+     **/
+    hdb_binsrch_info.hash_type = TSK_HDB_HTYPE_INVALID_ID;
+    // this value is read only in hdb_binsrch_idx_init_hash_type_info
+    hdb_binsrch_info.base.db_fname = (TSK_TCHAR *)db_name;
+    int ret_val = hdb_binsrch_idx_init_hash_type_info(&hdb_binsrch_info, htype);
+
+    REQUIRE(ret_val == expected_return);
+
+    if (expected_return == 0)
+    {
+        // if successful call, do comparisons
+        REQUIRE(hdb_binsrch_info.hash_len == expected_hash_len);
+        REQUIRE(TSTRCMP(hdb_binsrch_info.idx_fname, expected_idx_fname) == 0);
+        REQUIRE(TSTRCMP(hdb_binsrch_info.idx_idx_fname, expected_idx_idx_fname) == 0);
+    }
+
+    /**
+     * cleanup allocated items from hdb_binsrch_idx_init_hash_type_info.
+     * These are values set by hdb_binsrch_idx_init_hash_type_info during run 
+     * that need to be cleaned up. Normally, they would be cleaned up when the
+     * hash database is closed, but not for this unit test.
+     **/
+    if (hdb_binsrch_info.idx_fname != NULL)
```

Comment:
`free` is callable on `nullptr` and checks for nulls itself. There's no reason to check yourself whether the variable you're passing to `free` is nonnull first.

Reply:
This is so true, but many people think that calling `free(NULL)` will result in a null pointer error.